input_network_encoder.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "core/local_vector.h"
+#include "core/resource.h"
+#include "core/variant.h"
+#include "network_synchronizer/data_buffer.h"
+
+struct NetworkedInputInfo {
+	StringName name;
+	Variant default_value;
+	uint32_t index;
+	DataBuffer::DataType data_type;
+	DataBuffer::CompressionLevel compression_level;
+	real_t comparison_floating_point_precision;
+};
+
+class InputNetworkEncoder : public Resource {
+	GDCLASS(InputNetworkEncoder, Resource)
+
+public:
+	constexpr static uint32_t INDEX_NONE = UINT32_MAX;
+
+private:
+	LocalVector<NetworkedInputInfo> input_info;
+
+protected:
+	static void _bind_methods();
+
+public:
+	uint32_t register_input(
+			const StringName &p_name,
+			const Variant &p_default_value,
+			DataBuffer::DataType p_type,
+			DataBuffer::CompressionLevel p_compression_level,
+			real_t p_comparison_floating_point_precision = CMP_EPSILON);
+
+	uint32_t find_input_id(const StringName &p_name) const;
+	const LocalVector<NetworkedInputInfo> &get_input_info() const;
+
+	void encode(const LocalVector<Variant> &p_inputs, DataBuffer &r_buffer) const;
```

Comment:
possibly better name would be encode_inputs() (since I assume that's what we are encoding)

Reply:
I think it's getting insanely verbose at this point. I don't think it's going to make it any better:
```c++
encode(input, buffer);
// vs
encode_input(input, buffer);
```